File: TerminBA/TerminBA.Services/Database/WorkingHours.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Linq.Expressions;
using System.Text;
using System.Threading.Tasks;
using System.Xml;
using TerminBA.Models.Enums;

namespace TerminBA.Services.Database
{
    public class WorkingHours
    {
        [Key]
        public int Id { get; set; }

        [ForeignKey(nameof(SportCentar))]
        [Required]
        public int SportCenterId { get; set; }
        public SportCenter? SportCentar { get; set; }

        [Required]
        public DayOfWeek StartDay { get; set; }
        [Required]
        public DayOfWeek EndDay { get;set; }

        [Required]
        public TimeOnly OpeningHours { get; set; }
        [Required]
        public TimeOnly CloseingHours { get;set; }

        [Required]
        public DateOnly ValidFrom { get; set; }
        public DateOnly? ValidTo { get; set; } // can be null if the working hours are constant

        [NotMapped]
        public bool IsActive =>
            (ValidFrom <= DateOnly.FromDateTime(DateTime.Today)) &&
            (ValidTo == null || ValidTo >= DateOnly.FromDateTime(DateTime.Today));

        //public static Expression<Func<WorkingHours, bool>> IsActiveExpr(DateOnly today) =>
        // x => x.ValidFrom <= today &&
        //    (x.ValidTo == null || x.ValidTo >= today);
    }
}

File: TerminBA/TerminBA.UI/terminba_sport_center_desktop/lib/model/working_hours.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:terminba_sport_center_desktop/enums/day_of_week_enum.dart';

part 'working_hours.g.dart';

@JsonSerializable()
class WorkingHours {
	int id;
	int sportCenterId;
  @JsonKey(fromJson: dayOfWeekFromJson, toJson: dayOfWeekToJson)
  DayOfWeek startDay;
  @JsonKey(fromJson: dayOfWeekFromJson, toJson: dayOfWeekToJson)
  DayOfWeek endDay;
	String openingHours;
	String closeingHours;
	DateTime validFrom;
	DateTime? validTo;
	bool isActive;

	WorkingHours(this.id,this.sportCenterId,this.startDay,this.endDay,this.openingHours,this.closeingHours,this.validFrom,this.validTo,this.isActive,);

	factory WorkingHours.fromJson(Map<String, dynamic> json) => _$WorkingHoursFromJson(json);

	Map<String, dynamic> toJson() => _$WorkingHoursToJson(this);
}